File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(a, b)
    raise ArgumentError unless a.length == b.length
    (0..a.length).map{|x| a[x] == b[x] ? 0 : 1}.inject(:+)
  end
end

module BookKeeping
  VERSION = 3
end

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
  public static int compute(String a, String b) {
    int distance = 0;
    if (a.length() != b.length()) {
      throw new IllegalArgumentException();
    }
    for (int i = 0; i < a.length(); i++) {
      if (a.charAt(i) != b.charAt(i)) {
        distance++;
      }
    }
    return distance;
  }
}
